### PR TITLE
BF: Replace all occurrences of mutable default values with a safe default

### PR DESCRIPTION
**Pull request by hanke**
Separate PR to get a clean test of just that.

### CODE REVIEW

**Comment by nno** [comment]
Looks great, thanks.

File: mvpa2/atlases/base.py [hunk 1]
```diff
@@ -520,10 +522,12 @@ def __repr__(self):
 
     @staticmethod
     ##REF: Name was automagically refactored
-    def from_xml(Elevel, levelIndex=[0]):
+    def from_xml(Elevel, levelIndex=None):
         # XXX this is just for label type of level. For distance we need to ...
         # we need to assure the right indexing
 
+        if levelIndex is None:
+                levelIndex = [0]
```
Comment (yarikoptic) [inline mvpa2/atlases/base.py]: Over indebted?

File: mvpa2/base/node.py [hunk 2]
```diff
@@ -281,7 +281,9 @@ def set_postproc(self, node):
     def __str__(self, *args, **kwargs):
         return _str(self, *args, **kwargs)
 
-    def __repr__(self, prefixes=[]):
+    def __repr__(self, prefixes=None):
+        if prefixes is None:
+            prefixes = []
```
Comment (yarikoptic) [inline mvpa2/base/node.py]: It was perfectly fine before. More concise code and no chance for --inplace changes since +[]. Why to fix not broken?

**Comment by yarikoptic** [comment]
@nno could we not rush a bit as much?

**Comment by yarikoptic** [comment]
In this case I didn't like this one. Clearly many changes were due automatically and were not filtered out. In some places I would have preferred to keep it old way since it gives better idea on function signature, and had no side effect since that list was always added to another.  So why then to fix something not broken. Pep8 is not the law, it is a bible!

**Comment by hanke** [comment]
This is not about PEP8. This is about the fragility that keep mutable argument "perfectly fine".

**Comment by hanke** [comment]
In general those should be good and not hurt. Individual cases could be argued about, but then again: why make them special?

**Comment by yarikoptic** [comment]
Once again in many places it was perfectly fine and code was written to account for this effect. Did you find a bug? (Besides the one introduced by beautification)?

**Comment by yarikoptic** [comment]
Now many functions just look plain ugly

File: mvpa2/atlases/base.py (hunk 1 above)
Comment (hanke) [inline mvpa2/atlases/base.py]: yes, sorry. These files have more formating issues though...

File: mvpa2/base/node.py (hunk 2 above)
Comment (hanke) [inline mvpa2/base/node.py]: Speed accuracy trade-off. The fix is not that it was broken before. The fix is the function signature that looks more trustworthy now.

**Comment by hanke** [comment]
I don't want to argue. Feel free to revert. However, personally I'd rather have it ugly like this than having to investigate each and every time whether the function body still reflect the implications imposed by a mutuable argument.

File: mvpa2/base/node.py (hunk 2 above)
Comment (yarikoptic) [inline mvpa2/base/node.py]: I have no trust in function signatures. Moreover, the ones with all Nones in it defaults confuse me often

**Pull request merged**